pkg/samlcache/samlcache.go

```diff
@@ -0,0 +1,155 @@
+package samlcache
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"runtime"
+	"time"
+
+	homedir "github.com/mitchellh/go-homedir"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+var (
+	ErrInvalidCachePath = errors.New("Cannot evaluate Cache file path")
+	logger              = logrus.WithField("pkg", "samlcache")
+)
+
+const (
+	SAMLAssertionValidityTimeout = 5 * time.Minute
+	SAMLCacheFilePermissions     = 0600
+	SAMLCacheDirPermissions      = 0700
+	SAMLCacheDir                 = ".saml2aws_cache"
```

Comment:
AWS CLI is using [`.aws/cli/cache`](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-files.html) for its cache directory. If we want to use `.aws` directory then using something like `.aws/saml2aws/cache` would make more sense.

Reply:
no problem, that can be changed.
On the customization front, the code allows it but it's already a pretty dense PR and would prefer address this in a separate PR, if there's a need for it.